packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -705,7 +705,14 @@ class _CircularProgressIndicatorState extends State<CircularProgressIndicator> w
 
   Widget _buildCupertinoIndicator(BuildContext context) {
     final Color? tickColor = widget.backgroundColor;
-    return CupertinoActivityIndicator(key: widget.key, color: tickColor);
+    final double? val = widget.value;
+
+    if (val == null) {
+      return CupertinoActivityIndicator(key: widget.key, color: tickColor);
+    }
+
+    return CupertinoActivityIndicator.partiallyRevealed(
+        key: widget.key, color: tickColor, progress: val);
```

Comment:
Formatting nit: move each property to a new line since this line is already long:
```
return CupertinoActivityIndicator.partiallyRevealed(
     key: widget.key,
     color: tickColor,
     progress: val
);
```